### DISCUSSION TITLE
To get proxy host domain set in the mobile devices

### DISCUSSION THREAD

**Discussion by ashishmeher216** [mitmproxy/mitmproxy#6422] [Q&A]
#### Problem Description

I am trying to build a forward proxy server through which traffic from **mobile devices** can be captured and modified. My proxy server lies in an EC2 instance which is pointed by a wildcard domain, let's say ***.myproxy.com**. 

I am setting different proxy hosts  such as **abc.myproxy.com, xyz.myproxy.com**, etc in different mobile devices in their wifi configurations.  I am getting network traffic of the devices on my forward proxy but am **unable to distinguish** the requests according to the proxy host (abc.myproxy.com or xyz.myproxy.com). 

If a mobile device is hitting example.com from the browser then the request header's host contains example.com and their is **no special parameter in the request object** through which I can get the proxy host address set in that mobile device.

#### Proposal
I want to distinguish the network traffic of different devices according to the proxy host (abc.myproxy.com or xyz.myproxy.com) set in those devices. Is it possible to get the proxy host address set in the devices at my proxy server using MITM proxy?

**Comment by mhils** [comment]
For explicit HTTP proxying (https://docs.mitmproxy.org/stable/concepts-howmitmproxyworks/#explicit-http) this is not really possible. For explicit HTTPS you may be lucky and the client may be sending a Host header in the initial CONNECT? Those are available in the `http_connect` hook, see https://github.com/mitmproxy/mitmproxy/blob/10.1.1/mitmproxy/addons/proxyauth.py#L69-L72 for an example of how you can persist state.

Alternatively, can you just equip your clients with unique proxy authentication credentials?

**Comment by ashishmeher216** [comment]
Thanks for your response @mhils. No, I cannot use authentication credentials. Because the use case is that I can only inject the host and port of the proxy, which is abc.myproxy.com, and port number, let's say 8080.

But, if I point a static subdomain to the proxy server, let's say connect.myproxy.com. and pass host as connect.myproxy.com/path1, connect.myproxy.com/path2, etc as proxy host on different devices.
In this [discourse](https://discourse.mitmproxy.org/t/how-to-view-the-proxy-servers-url/23), there is a discussion about differentiating the proxy url based on their path. Is it possible with MITM?
